tests/subsys/portability/cmsis_rtos_v2/src/kernel.c

```diff
@@ -63,7 +63,20 @@ void lock_unlock_check(const void *arg)
 
 void test_kernel_apis(void)
 {
-	versionInfo version, version_irq;
+	versionInfo version = {
+		.os_info = {
+			.api = 0xfefefefe,
+			.kernel = 0xfdfdfdfd,
+		},
+		.info = "local function call info is uninitialized"
+	};
+	versionInfo version_irq = {
+		.os_info = {
+			.api = 0xfcfcfcfc,
+			.kernel = 0xfbfbfbfb,
+		},
+		.info = "irq_offload function call info is uninitialized"
+	};
 
 	get_version_check(&version);
```

Comment:
note: shouldn't this function return an error if `osKernelGetInfo` doesn't return `osOk`? test could then assert https://github.com/keith-packard/zephyr/blob/fix-cmsis_rtos_v2-test/tests/subsys/portability/cmsis_rtos_v2/src/kernel.c#L28

Reply:
It will fail the assertion of osKernelGetInfo doesn't succeed as the updated values won't be written into the return structure. This patch was not designed to fix the architecture of the test but rather fix a bug in the implementation of that design. If it were not test code, it might be a good idea to try to make it more understandable.